app/code/Magento/Store/Api/Data/StoreInterface.php

```diff
@@ -67,6 +67,17 @@ public function setWebsiteId($websiteId);
     public function getStoreGroupId();
 
     /**
+     * @param int $isActive
+     * @return $this
+     */
+    public function setIsActive($isActive);
```

Comment:
Unfortunately, introduction of any new methods to the `@api` interfaces is not possible due to [backwards compatibility policy](http://devdocs.magento.com/guides/v2.1/ext-best-practices/extension-coding/backwards-compatible-development/index.html). This change can potentially break any third-party customization which used this class as SPI (providing their own implementation for existing interface).

Possible workaround would be implementing an extension attribute for this class which would contain the required field. This would not break any existing customization but as a tradeoff would look not as good.

Reply:
this change was accepted by Anton even though it gave backward compliance issue. I am happy to give a go with the extension attribute but I'd say any developer would see this change as more beneficial as it is now.
--> what I mean is Magento 2 is complex enough to avoid making minor issue to become a big problem?